src/plugins/dashboard/public/dashboard_app/top_nav/editor_menu.tsx

```diff
@@ -41,6 +46,97 @@ interface UnwrappedEmbeddableFactory {
   isEditable: boolean;
 }
 
+export type GetEmbeddableFactoryMenuItem = ReturnType<typeof getEmbeddableFactoryMenuItemProvider>;
+
+export const getEmbeddableFactoryMenuItemProvider =
```

Comment:
Does this need to be a separate function? I see it's exported for test coverage, but I suspect it might be more robust to test this editor menu as a React component via `@testing-library/react`

Reply:
yes, I made it a separate function to test the functionality I added. 
It's indeed can be a jest test with `@testing-library/react`, but there were no tests for this file whatsoever. I want to keep changes to a minimum to unblock https://github.com/elastic/kibana/pull/179547

Reply:
Ideally, it should be covered with functional tests using FTR. I plan to add some in https://github.com/elastic/kibana/pull/179547, because first it needs this PR to get into `main`.